core/templates/domain/state/StateObjectFactorySpec.ts

```diff
@@ -94,6 +97,60 @@ describe('State Object Factory', () => {
     };
   });
 
+  beforeAll(() => {
+    constants.NEW_STATE_TEMPLATE = {
```

Comment:
Just curious, what happens if you do this without copying the value to `oldNewStateTemplate` and copying it back? Does it cause other tests to fail?

Reply:
It caused some tests to failed but only when not all tests were run. I think we should have some more general approach to mocking variables and constants.

Reply:
Oh there is a way to mock constants. Can you check out the usage of `$provide.constant(..)`? We seem to use that in multiple places.

Reply:
I think that this approach is only usable in AngularJS and not in Angular 8. Am I right here @bansalnitish?

Reply:
Maybe also @marianazangrossi, do you know if there is an approach how to mock Angular 8 constants in our frontend tests?

Reply:
Yes $provide.constant is not used in Angular, it was used in AngularJS.